shell: kernel_service: heap: add required header

ycsin opened the pull request:
Add the "kernel_shell.h" header which is required but missing, causing build error.

Fixes #78668

nashif commented on tests/subsys/shell/modules/kernel_service/src/main.c:
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (c) 2012-2014 Wind River Systems, Inc.
```
dislike this type of  build_only "test", especially if we have this already covered somewhere else, i.e. samples/subsys/shell/shell_module. Either add a new prj.conf there with additional options or make this a proper test please using pytest

ycsin replied:
done

ycsin replied:
Had some trouble getting the test right so I've dropped them, they will be done later in a separate PR

ycsin committed:
shell: kernel_service: heap: add required header

Add the "kernel_shell.h" header which is required but missing,
causing build error.

Signed-off-by: Yong Cong Sin <ycsin@meta.com>
Signed-off-by: Yong Cong Sin <yongcong.sin@gmail.com>

ycsin commented:
drop everything else leaving just the fix, the test needed more time and will be done in a separate PR
